sdcm/utils/aws_utils.py

```diff
@@ -268,8 +268,14 @@ def get_ec2_network_configuration(regions: list[str], availability_zones: list[s
     ec2_subnet_ids = []
     for region in regions:
         aws_region = AwsRegion(region_name=region)
+        if params.get('auto_availability_zone'):
+            availability_zones = aws_region.availability_zones
+        else:
+            availability_zones = [region + availability_zone for availability_zone in availability_zones]
+
+        LOGGER.debug("Getting subnets and SGs for availability zones of: %s", availability_zones)
         for availability_zone in availability_zones:
-            sct_subnet = aws_region.sct_subnet(region_az=region + availability_zone)
+            sct_subnet = aws_region.sct_subnet(region_az=availability_zone)
```

Comment:
I'm sorry, but I don't understand how it works or if it works as the requirement in the task.
(And if there is such a magic it should be explained in the commit message).

The requirement is to catch error of AZ that can't fullfill the provision and retry it on the other AZs.
What you did here is not specifying AZ?
Is there any promise that AWS will be able to provision the instances like this?
How is it even possible here that you first get the sunbet of the AZ even before you try to provision...

Reply:
fixed